flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/SpeculativeScheduler.java

```diff
@@ -85,6 +91,10 @@ public class SpeculativeScheduler extends AdaptiveBatchScheduler
 
     private final SlowTaskDetector slowTaskDetector;
 
+    private long numSlowExecutionVertices;
```

Comment:
How about change the name to `currentNumSlowExecutionVertices`? I think the `numXXX` generally represents an accumulated value (like `numEffectiveSpeculativeExecutions`).

Reply:
> I think the numXXX generally represents an accumulated value

Maybe not because we also have `numRunningJobs`, `numRegisteredTaskManagers`, etc. 
I do not have much preference here, but maybe we can stick to the proposed name in the FLIP?